### ISSUE TITLE
TCP hangs on FreeBSD

### ISSUE THREAD

**Issue by mneumann** [savi-lang/TCP#14]
I've seen this for `savi-Postgres` and now as well on `savi-MQTT`.

After sending a couple of packets, this can be 2, 50 or 100, it just hangs. 

Also, sometimes when running `savi spec`, the test suite hangs. Likely that the two issues are related.

The process is in Idle state.

To reproduce:

https://github.com/mneumann/savi-MQTT/

```
savi run example
```

You need a MQTT broker (mosquitto) running on localhost. (you can see that it is publishing by running `mosquitto_sub -t topic`

**Comment by jemc** [comment]
What gives me pause and makes me think this may be a different issue is that #2 happens on many file descriptors and gives an infinite loop.

**Comment by jemc** [comment]
> I am trying to build `libponyrt.bc` tomorrow myself.

I don't think this will be fruitful - it's likely that the Pony runtime version doesn't matter here.

What would be more helpful is some further info about how it hangs.

I'd suggest either using `lldb` or adding some `Inspect.out` statements to the TCP library here for some extra debugging to see if you can figure out what happens just before it hangs.

**Comment by mneumann** [comment]
It seems, it's unrelated to TCP. It also hangs when I just use a `Timer`. So I suspect it's the underlying event system.

So basically this hangs:

```savi
:actor Main
  :is Timer.Actor
  :let timer Timer.Engine

  :new (env)
    @timer = Timer.Engine.new(@, Time.Duration.milliseconds(1))

  :fun ref timer_react @
    elapsed = @timer.elapsed
    Inspect.out("TICK: \(elapsed.total_seconds.format.decimal).\(elapsed.nanosecond.format.decimal)")
    @
```

**Comment by mneumann** [comment]
Note that the timer example above does not seem to hang when I pass `--ponynoblock`. It consistently hangs without it. The full TCP example blocks regardless of `--ponynoblock`.

**Comment by jemc** [comment]
Haha okay I can move this issue back to the main repo then.

Do you get similar hangs with an equivalent Pony program?
